### DISCUSSION TITLE
ex23 lives

### DISCUSSION THREAD

**Discussion by gdmcbain** [kinnala/scikit-fem#1123] [Show and tell]
The Bratu&ndash;Gelfand turning-point example 
https://github.com/kinnala/scikit-fem/blob/c77b1dfd15c72112633ab15ddac70e5ebe40c60a/docs/examples/ex23.py#L1-L140 was deleted in 7c9dd7ad27049b3c94db841e0103e29c045568de under #910 a couple of years ago, but I recently stumbled across a scion: [demos/bratu-gelfand/bratu.py](https://gitlab.com/simon123h/bice/-/blob/593958ce942d35a58a608543e4d85dfb3d92d256/demos/bratu-gelfand/bratu.py).

This is in [bice](https://gitlab.com/simon123h/bice)

> A numerical path continuation software written in Python.

and their example is basically the old ex23 with bice replacing the other deprecated bifurcation package.

As I found it, it wasn't working with the latest skfem (getting dofs rather than finding them), but I've [fixed ](https://gitlab.com/simon123h/bice/-/commit/3ffdc27f60a9dcf346bd11346392fb2e0da274e2) that and now again there is an example of how to track the solutions of a steady nonlinear scikit-fem problem around a turning point.  I don't know whether that needs to be copied back here as a resurrected ex23, but I do note that `ex23.png` is still in `docs/examples/Makefile`.

https://github.com/kinnala/scikit-fem/blob/976f1cc8782f2e33e5e229b119d06ece63d8fb0c/docs/examples/Makefile#L10

The PNG produced by `bice/demons/bratu-gelfand/bratu.py` is:

![my_bifdiag](https://github.com/kinnala/scikit-fem/assets/1588947/f8f5eb83-1d33-455d-91d0-8671af1e7780)

**Comment by kinnala** [comment]
Interesting, thanks for the message!
